DFC.ServiceTaxonomy.PageLocation/Handlers/PageLocationPartHandler.cs

```diff
@@ -1,16 +1,56 @@
-﻿using System.Threading.Tasks;
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
 using DFC.ServiceTaxonomy.PageLocation.Models;
+using DFC.ServiceTaxonomy.PageLocation.Services;
 using OrchardCore.ContentManagement;
 using OrchardCore.ContentManagement.Handlers;
+using OrchardCore.ContentManagement.Records;
+using OrchardCore.Title.Models;
+using YesSql;
 
 namespace DFC.ServiceTaxonomy.PageLocation.Handlers
 {
     public class PageLocationPartHandler : ContentPartHandler<PageLocationPart>
     {
+        private readonly ISession _session;
+        private readonly IClonedPagePropertyGenerator _generator;
+
+        public PageLocationPartHandler(ISession session, IClonedPagePropertyGenerator generator)
+        {
+            _session = session;
+            _generator = generator;
+        }
+
         public override Task InitializingAsync(InitializingContentContext context, PageLocationPart part)
         {
             part.Apply();
             return Task.CompletedTask;
         }
+
+        public override async Task CloningAsync(CloneContentContext context, PageLocationPart part)
+        {
+            string title = context.CloneContentItem.Content[nameof(TitlePart)][nameof(TitlePart.Title)];
```

Comment:
This logic should be moved into `MutateOnClone` in `PageLocationPartGraphSyncer`.
Apart from it belonging there, if it stayed in it's own `ContentParthandler`, we'd be dependent on the order that the 2 content handlers that handle cloning get called in, for any mutations to the content item done here, to be synced to the graph(s).

Reply:
Done!